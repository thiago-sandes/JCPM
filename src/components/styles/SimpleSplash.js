import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  view: {
    width: '100%',
    height: '100%',
    backgroundColor: 'orange',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {color: 'white', fontSize: 40, fontWeight: 'bold', textAlign: 'center'},
  activityIndicator: {
    marginTop: '2%',
  },
});
