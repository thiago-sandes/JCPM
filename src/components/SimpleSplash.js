import React from 'react';
import {View, Text} from 'react-native';

// Styles
import styles from './styles/SimpleSplash';

const SimpleSplash = (props) => {
  return (
    <View style={styles.view}>
      <Text style={styles.text}>Carregando os dados...</Text>
    </View>
  );
};

export default SimpleSplash;
