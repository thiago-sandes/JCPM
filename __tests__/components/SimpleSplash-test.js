import React from 'react';
import renderer from 'react-test-renderer';
import SimpleSplash from '../../src/components/SimpleSplash';

jest.useFakeTimers();

function fakeTimer(callback) {
  setTimeout(() => {
    callback && callback();
  }, 1000);
}

test('SimpleSplash snapshot', async () => {
  const tree = renderer.create(<SimpleSplash />).toJSON();

  fakeTimer();

  expect(tree).toMatchSnapshot();
});
