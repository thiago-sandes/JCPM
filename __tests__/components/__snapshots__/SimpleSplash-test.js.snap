// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSplash snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "orange",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 40,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Carregando os dados...
  </Text>
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "marginTop": "2%",
      }
    }
  />
</View>
`;
